fsm.cpp

```diff
@@ -18,12 +20,29 @@ int state1_countdown_timer = 0;
 int state2_duration_timer = 0;
 int state3_stillness_timer = 0;
 
+std::queue<sensorData_t> sensorDataQueue;
+DoorSensor doorSensor = DoorSensor(DOOR_SENSOR_PIN);
+MotionSensor motionSensor = MotionSensor(MOTION_SENSOR_PIN);
+
 void fsm::setupFSM()
 {
     updateTimersFromFlash();
     DEBUG_SERIAL_LOG.printf("Timers constants: countdown: %u, duration: %u, stillness: %u\r\n", COUNTDOWN_TIMER, DURATION_TIMER, STILLNESS_TIMER);
 }
 
+int fsm::handleState()
+{
+    if (!sensorDataQueue.empty()) {
+        sensorData_t sensorData = sensorDataQueue.front();
+        sensorDataQueue.pop();
+        return stateHandler(sensorData);
+    }
+    sensorData_t sensorData = {
+        .doorOpen = doorSensor.isDoorOpen(),
+        .motion = motionSensor.isThereMotion()};
```

Comment:
Why do you need to read this here instead of just getting it off the queue?

Reply:
On second thought, this is fixable.